torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -508,7 +520,13 @@ struct CodeImpl {
           fmap(graph->outputs(), [](const Value* v) { return v->type(); }));
     }
     n_inputs = graph_->inputs().size();
-    // std::cout << *graph_ << "\n";
+  }
+
+  // since subclass of CodeImpl needs to populate
+  // op_to_num_specified_args, we seperate the calls
+  // that changes internals of CodeImpl into a separate
+  // function. User must manually call run() before using it
+  void run() {
```

Comment:
We should probably avoid this because the logic here used to be part of constructor, now users of Code need to call it manually. 

Could you try marking `run` as virtual, override it in `MobileCodeImpl` and call it as last step inside constructor? This would avoid an additional manual call to `run`

https://www.programiz.com/cpp-programming/virtual-functions

Reply:
done